en/docs/chapter_searching/searching_algorithm_revisited.md

```diff
@@ -55,23 +55,23 @@ The operation efficiency and characteristics of the aforementioned methods are s
 | Data preprocessing | /             | Sorting $O(n \log n)$ | Building tree $O(n \log n)$ | Building hash table $O(n)$ |
 | Data orderliness   | Unordered     | Ordered               | Ordered                     | Unordered                  |
 
-The choice of search algorithm also depends on the volume of data, search performance requirements, data query and update frequency, etc.
+The choice of search algorithm also depends on the volume of data, search performance requirements, frequency of data queries and updates, etc.
 
 **Linear search**
 
-- Good versatility, no need for any data preprocessing operations. If we only need to query the data once, then the time for data preprocessing in the other three methods would be longer than the time for linear search.
+- Good versatility, no need for any data preprocessing operations. If we only need to query the data once, then the time for data preprocessing in the other three methods would be longer than the time for a linear search.
 - Suitable for small volumes of data, where time complexity has a smaller impact on efficiency.
-- Suitable for scenarios with high data update frequency, because this method does not require any additional maintenance of the data.
+- Suitable for scenarios with very frequent data updates, because this method does not require any additional maintenance of the data.
 
 **Binary search**
 
-- Suitable for large data volumes, with stable efficiency performance, the worst time complexity being $O(\log n)$.
-- The data volume cannot be too large, because storing arrays requires contiguous memory space.
-- Not suitable for scenarios with frequent additions and deletions, because maintaining an ordered array incurs high overhead.
+- Suitable for larger data volumes, with stable performance and a worst-case time complexity of $O(\log n)$.
+- However, the data volume cannot be too large, because storing arrays requires contiguous memory space.
+- Not suitable for scenarios with frequent additions and deletions, because maintaining an ordered array incurs a lot of overhead.
 
```

Comment:
The first bullet point ('Suitable for large data volumes') seems to directly contradict the second bullet point ('The data volume cannot be too large') so it helps to clarify by adding 'However'.

(I imagine taking notes and seeing 'Suitable for large data volumes' so I write that down -> 'Good for large data'. Then I see 'The data volume cannot be too large' so I write that down -> 'Not good for large data'. With 'However' we clarify that there is a balance we need to be aware of.)